x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/action/RollupBeforeBulkInfo.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.rollup.action;
+
+import org.elasticsearch.common.io.stream.NamedWriteable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * This class includes statistics collected by the downsampling task before
+ * a bulk indexing operation starts.
+ */
+public class RollupBeforeBulkInfo implements NamedWriteable, ToXContentObject {
```

Comment:
I think this class, `RollupBulkInfo` and `RollupAfterBulkInfo` can be records instead of classes?